### ISSUE TITLE
Additional TAP APIService endpoints

### ISSUE THREAD

**Issue by siggy** [linkerd/linkerd2#3172]
PR #3167 introduced a Tap APIService with the following endpoints:
- GET /apis/tap.linkerd.io/v1alpha1
- POST /apis/tap.linkerd.io/v1alpha1/watch/namespaces/:ns/tap
- POST /apis/tap.linkerd.io/v1alpha1/watch/namespaces/:ns/:res/:name/tap

Kubernetes periodically queries for other endpoints, which are not yet implemented:
- GET /
```json
{
  "paths": [
    "/apis",
    "/apis/tap.linkerd.io/v1alpha1"
  ]
}
```
- GET /apis (APIGroupList?)
- GET /apis/tap.linkerd.io (APIGroup?)
- GET /openapi/v2
- GET /swagger.json (deprecated?)

Determine which additional endpoints the Tap APIService should support.

After generating an new project with https://github.com/kubernetes-incubator/apiserver-builder-alpha/blob/master/docs/tools_user_guide.md
```bash
$ curl -k https://localhost:9443
{
  "paths": [
    "/apis",
    "/apis/tap.linkerd.io",
    "/apis/tap.linkerd.io/v1alpha1",
    "/healthz",
    "/healthz/etcd",
    "/healthz/log",
    "/healthz/ping",
    "/metrics",
    "/openapi/v2",
    "/swagger-2.0.0.json",
    "/swagger-2.0.0.pb-v1",
    "/swagger-2.0.0.pb-v1.gz",
    "/swagger.json",
    "/version"
  ]
}
```

More info:
https://kubernetes.io/docs/concepts/overview/kubernetes-api/#openapi-and-swagger-definitions

**Comment by siggy** [comment]
```bash
$ curl -k https://localhost:9443/apis
{
  "kind": "APIGroupList",
  "groups": [
    {
      "name": "tap.linkerd.io",
      "versions": [
        {
          "groupVersion": "tap.linkerd.io/v1alpha1",
          "version": "v1alpha1"
        }
      ],
      "preferredVersion": {
        "groupVersion": "tap.linkerd.io/v1alpha1",
        "version": "v1alpha1"
      },
      "serverAddressByClientCIDRs": [
        {
          "clientCIDR": "0.0.0.0/0",
          "serverAddress": ":9443"
        }
      ]
    }
  ]
}
$ curl -k https://localhost:9443/apis/tap.linkerd.io
{
  "kind": "APIGroup",
  "apiVersion": "v1",
  "name": "tap.linkerd.io",
  "versions": [
    {
      "groupVersion": "tap.linkerd.io/v1alpha1",
      "version": "v1alpha1"
    }
  ],
  "preferredVersion": {
    "groupVersion": "tap.linkerd.io/v1alpha1",
    "version": "v1alpha1"
  }
}
$ curl -k https://localhost:9443/apis/tap.linkerd.io/v1alpha1
{
  "kind": "APIResourceList",
  "apiVersion": "v1",
  "groupVersion": "tap.linkerd.io/v1alpha1",
  "resources": [
    {
      "name": "deployments",
      "singularName": "",
      "namespaced": true,
      "kind": "Tap",
      "verbs": [
        "create",
        "delete",
        "deletecollection",
        "get",
        "list",
        "patch",
        "update",
        "watch"
      ]
    },
    {
      "name": "deployments/status",
      "singularName": "",
      "namespaced": true,
      "kind": "Tap",
      "verbs": [
        "create",
        "delete",
        "deletecollection",
        "get",
        "list",
        "patch",
        "update",
        "watch"
      ]
    },
    {
      "name": "deployments/tap",
      "singularName": "",
      "namespaced": true,
      "kind": "TapTap",
      "verbs": [
        "create",
        "get",
        "patch",
        "update"
      ]
    }
  ]
}
```

**Comment by grampelberg** [comment]
@siggy one of these has a kind of `TapTap`, is that a bug?

**Comment by grampelberg** [comment]
Shouldn't the verbs for `https://localhost:9443/apis/tap.linkerd.io/v1alpha1` be limited to only `watch`?

**Comment by siggy** [comment]
@grampelberg Yeah, that `TapTap` and those extra verbs are coming from a generated service based on https://github.com/kubernetes-incubator/apiserver-builder-alpha. 

The implementation in #3167 correctly has `Tap`, and also lists only `watch` as a verb. The other APIs recently added were reverse-engineered from based on https://github.com/kubernetes-incubator/apiserver-builder-alpha ;)

**Comment by grampelberg** [comment]
Well, at least `alpha` is in the name =)

**Comment by siggy** [comment]
Fixed via #3167.
